azurerm/resource_arm_postgresql_virtual_network_rule.go

```diff
@@ -165,7 +138,14 @@ func resourceArmPostgreSQLVirtualNetworkRuleRead(d *schema.ResourceData, meta in
 	d.Set("server_name", serverName)
 
 	if props := resp.VirtualNetworkRuleProperties; props != nil {
-		d.Set("subnet_id", props.VirtualNetworkSubnetID)
+
+		if virtualNetworkSubnetId := props.VirtualNetworkSubnetID; virtualNetworkSubnetId != nil {
+			d.Set("subnet_id", virtualNetworkSubnetId)
+		}
+
+		if ignoreMissingVnetServiceEndpoint := props.IgnoreMissingVnetServiceEndpoint; ignoreMissingVnetServiceEndpoint != nil {
+			d.Set("ignore_missing_vnet_service_endpoint", ignoreMissingVnetServiceEndpoint)
+		}
```

Comment:
we don't need to nil-check either of these - d.Set will handle that for us (since this is a simple type) - can we revert this?

Reply:
Yes, I can revert it no problem, but I did it cause in the SDK struct they are pointers and I was afraid that they may come back `nil` if something went wrong on the `read`

```
// VirtualNetworkSubnetID - The ARM resource id of the virtual network subnet.
VirtualNetworkSubnetID *string `json:"virtualNetworkSubnetId,omitempty"`

// IgnoreMissingVnetServiceEndpoint - Create firewall rule before the virtual network has vnet service endpoint enabled.
IgnoreMissingVnetServiceEndpoint *bool `json:"ignoreMissingVnetServiceEndpoint,omitempty"`
```